src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkWatchDog.java

```diff
@@ -61,38 +61,39 @@ public ZkWatchDog(String id, String serviceName, String connectionString,
             signal = Signal.NONE;
         }
         this.signal = signal;
-        initWatch();
     }
 
     @Override
-    void validateNodes() throws KeeperException, InterruptedException {
+    void validateNodes() throws KeeperException, InterruptedException, IOException {
         super.validateNodes();
         ensureZNode(serviceName, id, SIGNAL);
         ensureZNode(DEFAULT_BUILD_VERSION.getBytes(), serviceName, id, BUILD_VERSION);
+        nodesValidated = true;
     }
 
 
     @VisibleForTesting
-    void subscribeSignal() throws KeeperException, InterruptedException {
+    void subscribeSignal() {
         checkData(signalPath);
     }
 
     @VisibleForTesting
-    void subscribeBuildVersion() throws KeeperException, InterruptedException {
+    void subscribeBuildVersion() {
         checkData(buildVersionPath);
     }
 
-    private void checkData(String path) throws KeeperException, InterruptedException {
+    private void checkData(String path) {
         zookeeper.getData(path, this, this, null);
     }
 
     @Override
-    void initWatch() {
+    public void init() {
         try {
+            initZk();
```

Comment:
It must be the responsibility of ZkClient...

Reply:
`initZk` creates connection to ZK. The whole idea of this patch is to split creating of ZkClient instance and connecting to ZK.